lib/AST/ASTPrinter.cpp

```diff
@@ -4043,15 +4069,472 @@ void PrintAST::visitMissingMemberDecl(MissingMemberDecl *decl) {
+}
+
+void PrintAST::visitDotSelfExpr(DotSelfExpr *expr) {
+  visit(expr->getSubExpr());
+  Printer << ".self";
+}
+
+void PrintAST::visitErasureExpr(ErasureExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitKeyPathExpr(KeyPathExpr *expr) {
+}
+
+void PrintAST::visitForceTryExpr(ForceTryExpr *expr) {
+  Printer << "try! ";
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitSequenceExpr(SequenceExpr *expr) {
+}
+
+void PrintAST::visitSuperRefExpr(SuperRefExpr *expr) {
+}
+
+void PrintAST::visitMemberRefExpr(MemberRefExpr *expr) {
+  visit(expr->getBase());
+  Printer << ".";
+  Printer << expr->getMember().getDecl()->getName();
+}
+
+void PrintAST::visitSubscriptExpr(SubscriptExpr *expr) {
+}
+
+void PrintAST::visitEnumIsCaseExpr(EnumIsCaseExpr *expr) {
+}
+
+void PrintAST::visitForceValueExpr(ForceValueExpr *expr) {
+}
+
+void PrintAST::visitKeyPathDotExpr(KeyPathDotExpr *expr) {
+}
+
+void PrintAST::visitAutoClosureExpr(AutoClosureExpr *expr) {
+  visit(expr->getSingleExpressionBody());
+}
+
+void PrintAST::visitCaptureListExpr(CaptureListExpr *expr) {
+}
+
+void PrintAST::visitDynamicTypeExpr(DynamicTypeExpr *expr) {
+}
+
+void PrintAST::visitOpaqueValueExpr(OpaqueValueExpr *expr) {
+}
+
+void PrintAST::visitOptionalTryExpr(OptionalTryExpr *expr) {
+}
+
+void PrintAST::visitPrefixUnaryExpr(PrefixUnaryExpr *expr) {
+}
+
+void PrintAST::visitBindOptionalExpr(BindOptionalExpr *expr) {
+}
+
+void PrintAST::visitBridgeToObjCExpr(BridgeToObjCExpr *expr) {
+}
+
+void PrintAST::visitObjCSelectorExpr(ObjCSelectorExpr *expr) {
+}
+
+void PrintAST::visitPostfixUnaryExpr(PostfixUnaryExpr *expr) {
+}
+
+void PrintAST::visitTupleElementExpr(TupleElementExpr *expr) {
+}
+
+void PrintAST::visitDerivedToBaseExpr(DerivedToBaseExpr *expr) {
+}
+
+void PrintAST::visitDotSyntaxCallExpr(DotSyntaxCallExpr *expr) {
+  visit(expr->getBase());
+  Printer << ".";
+  visit(expr->getFn());
+}
+
+void PrintAST::visitObjectLiteralExpr(ObjectLiteralExpr *expr) {
+}
+
+void PrintAST::visitUnresolvedDotExpr(UnresolvedDotExpr *expr) {
+  visit(expr->getBase());
+  Printer << ".";
+  Printer << expr->getName().getBaseName();
+}
+
+void PrintAST::visitArrayToPointerExpr(ArrayToPointerExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitBridgeFromObjCExpr(BridgeFromObjCExpr *expr) {
+}
+
+void PrintAST::visitCodeCompletionExpr(CodeCompletionExpr *expr) {
+}
+
+void PrintAST::visitInOutToPointerExpr(InOutToPointerExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitLinearFunctionExpr(LinearFunctionExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitDefaultArgumentExpr(DefaultArgumentExpr *expr) {
+}
+
+void PrintAST::visitLazyInitializerExpr(LazyInitializerExpr *expr) {
+  visit(expr->getSubExpr());
+}
+
+void PrintAST::visitOpenExistentialExpr(OpenExistentialExpr *expr) {
+  visit(expr->getValueProvidingExpr());
+  visit(expr->getExistentialValue());
+  visit(expr->getSubExpr());
```

Comment:
I believe just `visit(expr->getSubExpr())` would be sufficient here, could you add a test to make sure a method call on a protocol-typed value gets printed correctly?

Reply:
Looks like we do need `visit(expr->getExistentialValue())`, but not `visit(expr->getValueProvidingExpr())`.
Added test to verify.